diff --git a/src/ffmpeg.mk b/src/ffmpeg.mk
index 8dd80d86..e2132c3c 100644
--- a/src/ffmpeg.mk
+++ b/src/ffmpeg.mk
@@ -8,9 +8,9 @@ $(PKG)_CHECKSUM := cf38e0e28c7e5605942c4a77755349b0145804a397af37eb1fb4c77cb237f
 $(PKG)_SUBDIR   := $(PKG)-$($(PKG)_VERSION)
 $(PKG)_FILE     := $(PKG)-$($(PKG)_VERSION).tar.xz
 $(PKG)_URL      := https://ffmpeg.org/releases/$($(PKG)_FILE)
-$(PKG)_DEPS     := cc bzip2 gnutls lame libass libbluray libbs2b libcaca \
-                   libvpx opencore-amr $(BUILD)~nasm opus sdl2 speex theora vidstab \
-                   vo-amrwbenc vorbis x264 x265 xvidcore zlib
+$(PKG)_DEPS     := cc bzip2 \
+                   libvpx opencore-amr $(BUILD)~nasm opus sdl2 speex \
+                   vo-amrwbenc vorbis x264 x265 zlib
 
 # DO NOT ADD fdk-aac OR openssl SUPPORT.
 # Although they are free softwares, their licenses are not compatible with
@@ -42,24 +42,35 @@ define $(PKG)_BUILD
         --enable-gpl \
         --enable-version3 \
         --extra-libs="`'$(TARGET)-pkg-config' --libs libmpg123` -mconsole" \
-        --enable-gnutls \
-        --enable-libass \
-        --enable-libbluray \
-        --enable-libbs2b \
-        --enable-libcaca \
-        --enable-libmp3lame \
         --enable-libopencore-amrnb \
         --enable-libopencore-amrwb \
         --enable-libopus \
         --enable-libspeex \
-        --enable-libtheora \
-        --enable-libvidstab \
         --enable-libvo-amrwbenc \
-        --enable-libvorbis \
         --enable-libvpx \
         --enable-libx264 \
         --enable-libx265 \
-        --enable-libxvid \
+        --disable-programs \
+        --enable-protocol=file \
+        --enable-muxer=matroska \
+        --enable-muxer=mjpeg \
+        --enable-demuxer=mjpeg \
+        --enable-decoder=h264 \
+        --enable-decoder=hevc \
+        --enable-decoder=libvpx-vp9 \
+        --enable-decoder=mjpeg \
+        --enable-decoder=opus \
+        --enable-decoder=libopus \
+        --enable-decoder=pcm_alaw \
+        --enable-decoder=g722 \
+        --enable-encoder=mjpeg \
+        --enable-encoder=libvpx-vp9 \
+        --enable-encoder=libx264 \
+        --enable-encoder=libx265 \
+        --enable-encoder=opus \
+        --enable-encoder=libopus \
+        --enable-encoder=pcm_alaw \
+        --enable-encoder=g722 \
         --extra-ldflags="-fstack-protector" \
         $($(PKG)_CONFIGURE_OPTS)
     $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)'
